test/components/preprocessors/test_embedding_based_document_splitter.py

```diff
@@ -0,0 +1,487 @@
+        mock_embedder.run.assert_called_once()
+
+    def test_to_dict(self):
+        mock_embedder = Mock()
+        mock_embedder.to_dict.return_value = {"type": "MockEmbedder"}
+
+        splitter = EmbeddingBasedDocumentSplitter(
+            document_embedder=mock_embedder,
+            sentences_per_group=2,
+            percentile=0.9,
+            min_length=50,
+            max_length=1000,
+        )
+
+        result = splitter.to_dict()
+
+        assert "EmbeddingBasedDocumentSplitter" in result["type"]
+        assert result["init_parameters"]["sentences_per_group"] == 2
+        assert result["init_parameters"]["percentile"] == 0.9
+        assert result["init_parameters"]["min_length"] == 50
+        assert result["init_parameters"]["max_length"] == 1000
+        assert "document_embedder" in result["init_parameters"]
+
+
+    @pytest.mark.integration
+    def test_split_document_with_multiple_topics(self):
+        import torch
+        import os
+
+        # Force CPU usage to avoid MPS memory issues
+        os.environ["PYTORCH_ENABLE_MPS_FALLBACK"] = "1"
+        torch.backends.mps.is_available = lambda: False
+
+        embedder = SentenceTransformersDocumentEmbedder(
+            model="sentence-transformers/all-MiniLM-L6-v2",
+            device=ComponentDevice.from_str("cpu"),
+        )
+
+        embedder.warm_up()
+
+        splitter = EmbeddingBasedDocumentSplitter(
+            document_embedder=embedder,
+            sentences_per_group=2,
+            percentile=0.9,
+            min_length=30,
+            max_length=300,
+        )
+        splitter.warm_up()
+
+        # A document with multiple topics
+        text = (
+            "The weather today is beautiful. The sun is shining brightly. The temperature is perfect for a walk. "
+            "Machine learning has revolutionized many industries. Neural networks can process vast amounts of data. "
+            "Deep learning models achieve remarkable accuracy on complex tasks. "
+            "Cooking is both an art and a science. Fresh ingredients make all the difference. "
+            "Proper seasoning enhances the natural flavors of food. "
+            "The history of ancient civilizations fascinates researchers. Archaeological discoveries reveal new insights. "
+            "Ancient texts provide valuable information about past societies."
+        )
+        doc = Document(content=text)
+
+        result = splitter.run(documents=[doc])
+        split_docs = result["documents"]
+
+        # There should be more than one split
+        assert len(split_docs) > 1
+        # Each split should be non-empty and respect min_length
+        for split_doc in split_docs:
+            assert split_doc.content.strip() != ""
+            assert len(split_doc.content) >= 30
+        # The splits should cover the original text
+        combined = " ".join([d.content for d in split_docs]).replace(" ", "")
+        original = text.replace(" ", "")
```

Comment:
I think it'd better to rewrite the test to this. Otherwise it feels a bit like cheating since we remove all white space in each text which is a bit odd. I tested locally and this version still passes for me
```suggestion
        combined = "".join([d.content for d in split_docs])
        original = text
```